### PR TITLE
BF/TYPE: simplify some type checks; fix shell completion bug

### PR DESCRIPTION
**Pull request by aqw**
This PR has two unrelated fixes

- simplifies some unnecessary (and actively annoyingly) specific type annotations (#665)
- silences a one-time user error when first using shell completion (#585)

### CODE REVIEW

**Comment by bpoldrack** [comment]
I'm kinda  fine with that as is, but I'd like to point out, that I changed the type annotation for most things in PR #668 to `dict | UserDict` - not even restricting it to `Dict[str, Any]`. That's because most of the functions would work perfectly fine with any key type. It does not currently make sense to actually use other keys, but to me that's a matter of testing and it's only really relevant when loading/writing YAML. And even then: We'd simply get an exception, whenever ruamel doesn't know how to represent that key - no need to cover that by restrictive type annotation IMO.

So, in short: I'd be even more "loose" here (`dict | UserDict`) unless we know specifically that a particular function actually cares for the keys being strings.

**Comment by aqw** [comment]
I'm comfortable with that. Can do. :-)
